python/llama-index-server/llama_index/server/api/routers/chat.py

```diff
@@ -47,10 +54,23 @@ async def chat(
                 workflow = workflow_factory(chat_request=request)
             else:
                 workflow = workflow_factory()
-            workflow_handler = workflow.run(
-                user_msg=user_message.content,
-                chat_history=chat_history,
-            )
+
+            # Check if we should resume a chat with a human response
+            human_response = last_message.human_response
+            if human_response:
+                previous_ctx = WorkflowService.load_context(request.id, workflow)
+                # send a new human response event then resume the workflow with the previous context
+                previous_ctx.send_event(
+                    HumanResponseEvent(
+                        response=human_response,
+                    )
+                )
```

Comment:
try to separete persistence concern and HITL - if `request.id` is sent, we should re-create the workflow

```suggestion
            if request.id:
                previous_ctx = WorkflowService.load_context(request.id, workflow)
                # send a new human response event then resume the workflow with the previous context
```


then we need to run the workflow with a specific event, either start event as is or the human response event